core/src/replay_stage.rs

```diff
@@ -665,6 +674,10 @@ impl ReplayStage {
             let blockhash = bank.last_blockhash();
             vote_tx.partial_sign(&[node_keypair.as_ref()], blockhash);
             vote_tx.partial_sign(&[voting_keypair.as_ref()], blockhash);
+            // before publishing the latest vote on the network, snapshot the tower for persistent save
+            if let Err(e) = Self::snapshot_tower(&tower_snapshot_path, &tower) {
+                error!("Unable to backup tower, {:?}", e);
```

Comment:
@sagar-solana @TristanDebrunner, if you can't snapshot tower you should probably panic